libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
What you are proposing essentially turns the DI container into a service locator (which is a less-than-favorable practice in itself) and the way you are doing it increases coupling to a very specific structure.  Using the pull request you sent to the Framework's Application package as an example, you have deprecated everything that is dependency injected within the application class chain in favor of storing everything within the container (including the `stdClass $response` object used by the web application class) and have coded the application so that it is impossible to inject a DI container into that class chain.  You also bound the `Joomla\Registry\Registry` class solely to the application's configuration object which means the DI container cannot be used for instantiating classes implementing the `JModel` interface (in our implementations one of the constructor arguments is a `Registry` instance representing the model's state), a `JHttp` adapter (its $options argument in the CMS still requires a `Registry` object, this coupling is loosened a bit in the Framework), or any other class accepting a `Registry` object in its constructor; that argument will ALWAYS be the application configuration as you've bound it.

Yes, the front controller scripts I've shared are rather verbose.  They truthfully aren't much longer than the bootup process the CMS goes through (remember each index.php file also includes a defines.php and framework.php file before the app is instantiated; mine is all done in one file with a fair bit of error handling logic too).  But, those front controller scripts also support the notion of proper dependency injection throughout the full application infrastructure (and yes, the application needs to be held in the container since the `JController` interface expects controllers to have an application and input object, so for my service providers to be able to build those objects the container must be aware of the application).  Your proposal as structured basically makes it impossible to do anything with injected dependencies.